packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -147,6 +150,40 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform  {
             .contains('FlutterGeneratedPluginSwiftPackage');
   }
 
+  /// True if this project doesn't have Swift Package Manager disabled in the
+  /// pubspec, has either an iOS or macOS platform implementation, is not a
+  /// module project, Xcode is 15 or greater, and the Swift Package Manager
+  /// feature is enabled.
+  bool get usesSwiftPackageManager {
+    if (!featureFlags.isSwiftPackageManagerEnabled) {
+      return false;
+    }
+
+    // The project can disable Swift Package Manager in its pubspec.yaml.
+    if (parent.manifest.disabledSwiftPackageManager) {
+      return false;
+    }
+
+    // TODO(loic-sharma): Support Swift Package Manager in add-to-app modules.
+    // https://github.com/flutter/flutter/issues/146957
+    if (parent.isModule) {
+      return false;
+    }
+
+    if (!existsSync()) {
+      return false;
+    }
+
+    // Swift Package Manager requires Xcode 15 or greater.
```

Comment:
It might be time to require Xcode 15, since Apple has required it for iOS App Store submissions since April https://github.com/flutter/flutter/issues/144582.  See https://github.com/flutter/flutter/pull/146367.



https://developer.apple.com/ios/submit/
> All iOS and iPadOS apps uploaded to App Store Connect must be built with a minimum of Xcode 15 and the iOS 17 SDK.